auslib/admin/views/scheduled_changes.py

```diff
@@ -149,13 +146,27 @@ def get(self, sc_id):
             .execute()\
             .fetchone()[0]
 
+        # Although Scheduled Changes are stored across two tables, we don't
+        # expose that through the API. Because of this, we need to look up
+        # history in both and return the combined version.
+        # This is done by the database layer for non-history parts of Scheduled Changes, but
+        # that's not feasible for History due to the inheritance structure of the tables,
+        # so we do it here instead.
         revisions = self.table.scheduled_changes.history.select(
             where=[self.table.scheduled_changes.history.sc_id == sc_id,
                    self.table.scheduled_changes.history.data_version != null()],
             limit=limit,
             offset=offset,
             order_by=[self.table.scheduled_changes.history.timestamp.asc()],
         )
+        # There's a big 'ol assumption here that the primary Scheduled Changes
+        # table and the conditions table always keep their data version in sync.
```

Comment:
I guess there's two things that I'm concerned about. One is that I've done something wrong in this patch, and there's a case where they will get out of sync as part of the normal operations. The other is that some future code change may change things in such a way that normal operations would cause them to get out of sync. Raising an execption is actually a clever idea - if you do the data_version checking within the transaction, and raise on mismatch, the transaction will be aborted before it gets committed. It's still not a great user experience, but it's better than letting the db get into a bad state?

Manual db changes or some sort of crazy problem with transactions certainly could break things as well, but I'm not sure I'm terribly concerned about them. They're pretty close to "everything is on fire" territory for me, I guess.